python_modules/dagster/dagster/_daemon/sensor.py

```diff
@@ -659,20 +660,28 @@ def _is_under_min_interval(state, external_sensor):
     return elapsed < external_sensor.min_interval_seconds
 
 
-def _fetch_existing_runs(instance, external_sensor, run_requests):
+def _fetch_existing_runs(
+    instance,
+    external_sensor: ExternalSensor,
+    run_requests,
+):
     run_keys = [run_request.run_key for run_request in run_requests if run_request.run_key]
 
     if not run_keys:
         return {}
 
-    runs_with_run_keys = instance.get_runs(filters=RunsFilter(tags={RUN_KEY_TAG: run_keys}))
-
-    valid_runs = [
-        run for run in runs_with_run_keys if run.tags.get(SENSOR_NAME_TAG) == external_sensor.name
-    ]
+    runs_with_run_keys = instance.get_runs(
+        filters=RunsFilter(
+            tags={
+                RUN_KEY_TAG: run_keys,
```

Comment:
Worried about perf implications of adding additional tags as per data from the thread I posted earlier

also worried that this could filter out old runs from before the repo label tag existed, and suddenly cause a bunch of new runs to be created that has run keys already 


To fix both worries, could we do this filtering in memory instead using the external_pipeline_origin field on the candidate run? That has the nice property of still always being set for runs from before the repo label tag was created. Would still need to check the sensor name tag most likely (but doing that in memory as well perhaps?)